src/lib/plugins/chat.js

```diff
@@ -1,10 +1,122 @@
 module.exports.server=function(serv)
 {
-  serv.broadcast = (message, color) =>
-    serv.players.forEach(player => player.chat({
-      "text": message,
-      "color": color
-    }));
+  serv.broadcast = (message, {whitelist=serv.players,blacklist=[],system=false}={}) => {
+    if (whitelist.type == 'player') whitelist = [whitelist];
+
+    if (typeof message == 'string') message = serv.parseClassic(message);
+
+    whitelist.filter(w => blacklist.indexOf(w) == -1).forEach(player => {
+      if (!system) player.chat(message);
+      else player.system(message);
+    });
+  }
+
+  serv.color = {
+    'black': '&0',
+    'dark_blue': '&1',
+    'dark_green': '&2',
+    'dark_cyan': '&3',
+    'dark_red': '&4',
+    'purple': '&5',
+    'dark_purple': '&5',
+    'gold': '&6',
+    'gray': '&7',
+    'grey': '&7',
+    'dark_gray': '&8',
+    'dark_grey': '&8',
+    'blue': '&9',
+    'green': '&a',
+    'aqua': '&b',
+    'cyan': '&b',
+    'red': '&c',
+    'pink': '&d',
+    'light_purple': '&d',
+    'yellow': '&e',
+    'white': '&f',
+    'random': '&k',
+    'obfuscated': '&k',
+    'bold': '&l',
+    'strikethrough': '&m',
+    'underlined': '&n',
+    'underline': '&n',
+    'italic': '&o',
+    'italics': '&o',
+    'reset': '&r'
+  };
+
+  serv.parseClassic = (message) => {
+    if (typeof message == 'object') return message;
+    var messageList = [];
+    var text = '';
+    var nextChanged = false;
+    var color = 'white';
+    var bold = false;
+    var italic = false;
+    var underlined = false;
+    var strikethrough = false;
+    var random = false;
+    var colors = ['0','1','2','3','4','5','6','7','8','9','a','b','c','d','e','f', 'k','l','m','n','o','r','&']
+    var convertColor = ['black', 'dark_blue','dark_green','dark_cyan','dark_red','dark_purple','gold',
+                        'gray', 'dark_gray', 'blue', 'green', 'aqua', 'red', 'light_purple', 'yellow', 'white',
+                        'random', 'bold', 'strikethrough', 'underlined', 'italic', 'reset', '&'];
+
+    function createJSON() {
+      if (!text.trim()) return;
+      messageList.push({
+        text: text,
+        color: color,
+        bold: bold,
+        italic: italic,
+        underlined: underlined,
+        strikethrough: strikethrough,
+        obfuscated: random
```

Comment:
fyi, `{text,color,bold,italic,underlined,strikethrough,obfuscated}` works in es6